### PR TITLE
Add ISSUE_TEMPLATE for github

### PR DESCRIPTION
**Pull request by fendor**
The purpose would be to reduce the amount of back and forths we usually ask before we can properly answer a question. 
It might be overkill to ask that much information.

### CODE REVIEW

**Comment by Kleidukos** [comment]
Pretty sure we could have a tool that would produce a report for some of those values.

```
* Version of haskell-language-server
* Version of cabal
* Version of ghc
* Version of stack
```
These ones are fairly easy to automatically get.

**Comment by fendor** [comment]
Indeed, we could incorporate some of these into `haskell-language-server` themselves. 
Maybe we should also mention `haskell-language-server-wrapper` as well?

**Comment by fendor** [comment]
Would we re-invent the wheel if we add something like a `--prove-tools` command for hls?

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+* Version of ghc
+* Version of stack
+* Which lsp-client do you use
+  * Neovim, emacs, VS Codium, etc...
+* Describe your project (alternative: link to the project)
+  * Include `stack.yaml`
+  * Include `package.yaml`
+  * Include `*.cabal` files
+  * Include `cabal.project`
+* Contents of `hie.yaml` (See [Project Configuration](https://github.com/haskell/haskell-language-server#project-configuration) if you don't know what this is)
```
Comment (bubba) [inline .github/ISSUE_TEMPLATE.md:18]: Nitpick, probably don't need the "if you don't know what this is" part

**Comment by bubba** [comment]
Probably the most sensible thing to do would just be to print the versions/which tools are installed to stderr when `haskell-language-server[-wrapper]` is run

**Comment by fendor** [comment]
@bubba I am preparing a patch.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+<details>
+<summary>
+Lsp logs:
+</summary>
+
+```
+<paste your logs here>
+```
+</details>
```
Comment (Ailrun) [inline .github/ISSUE_TEMPLATE.md:49]: Could you put the final line feed?

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+If you encounter a bug or you have a support question, please include the following information:
+
+### Subject of the issue
+Describe your issue here.
+
+### Your environment
+* Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`
```
Comment (fendor) [inline .github/ISSUE_TEMPLATE.md:7]: Maybe we should mention that this cli flag is not available in the latest release? Or wait till the next release?

**Comment by fendor** [comment]
@alanz While I kind of agree, we also need that kind of information if there are bugs. Even on a private code-base, we can not help or debug at all, if not at least big parts of these informations are given. However, I agree that we can make it more clear that most of it is optional and that we dont close issues just because they dont follow that template.

**Comment by alanz** [comment]
@fendor I agree, but I think raising the issue, and giving the environment can be useful in some cases. And perhaps we should say that if they can repro do that, but if not still report.

Some bugs are straightforward to fix, without the repro, no need to put people through that pain if it is not needed.

File: .github/ISSUE_TEMPLATE.md (hunk 3 above)
Comment (Ailrun) [reply .github/ISSUE_TEMPLATE.md:7]: I think even after we release the next version, this template should mention the version range in which this command is available. If not, some users will try to run those command with outdated HLS, and then may think this issue template is outdated (or somewhat broken) and ignore the template.

**Pull request merged**